### ISSUE TITLE
Link from HACKING.adoc is wrong

### ISSUE THREAD

**Issue by johnwhitington** [gasche/opam-compiler-conf#15]
HACKING.adoc in the OCaml compiler says:

```
5. Install in a new opam switch to try things out:
+
----
opam compiler-conf install
----
```

But it appears it should, instead, point to this script.

**Comment by johnwhitington** [comment]
Please ignore. Did not read help properly.

**Comment by johnwhitington** [comment]
In fact, it seems `opam-compiler-conf.sh` no longer works. After running `configure` and `make world.opt`, I then get:

````
gorge:ocaml john$ opam-compiler-conf.sh install
Error: You must run the 'configure' command first.
```

(This is on trunk OCaml as of today)

**Comment by gasche** [comment]
Yes, opam-compiler-conf currently doesn't work with opam2, sorry. See #14.

**Comment by gasche** [comment]
I clarified the README to indicate this clearly: 52689c6ae17dc763ca7b1aeba0cd210f01e965f8. Thanks for the notice!

**Comment by johnwhitington** [comment]
Thanks!

Do you have a suggestion for a workflow for hacking the compiler now? I want to modify the OCaml compiler, but use it with OPAM (because I want to write a program using the modified compiler-libs and install packages on top).

Before OPAM 2.0, I could just install the new compiler as the system one, and then use "opam switch system", but "opam switch system" now only works for system compilers which are actual release compilers through "opam switch create ocaml-system" (and that doesn't work at the moment, it seems https://github.com/ocaml/opam/issues/3665)

Should I just abandon OPAM for this purpose, and return to using plain ocamlfind to install packages?

**Comment by gasche** [comment]
No, you should definitely keep using OPAM! Various workflows are detailed in https://github.com/ocaml/opam/issues/2531. A very simple one (doing more recompilation than `opam-compiler-conf` does, but that's a relatively small price to pay) was suggested by @Drup. From a trunk directory:

```
% opam switch install 4.03.0+trunk 
% opam pin add ocaml source/ocaml
```

4.03.0+trunk was the name of the development version at the time -- we are getting old! If I remember correctly, the +trunk switches for not-yet-released OCaml versions are not available on the default opam-repository anymore, you need to add a specific repository for that -- I think that <https://github.com/ocaml/ocaml-pr-repository> is currently the place to find these? (cc @avsm)
